Sources/NIO/BaseSocket.swift

```diff
@@ -301,6 +301,29 @@ class BaseSocket: BaseSocketProtocol {
         }
         return sock
     }
+    
+    /// Cleanup the unix domain socket.
+    ///
+    /// Deletes the associated file if it exists and has socket type.
+    ///
+    /// - parameters:
+    ///     - unixDomainSocketPath: The pathname of the UDS.
+    /// - throws: An `IOError` if the operation failed.
+    static func cleanupSocket(unixDomainSocketPath: String) throws {
+        let sb: UnsafeMutablePointer<stat> = UnsafeMutablePointer<stat>.allocate(capacity: 1)
```

Comment:
This pointer is leaked because it's never freed.

In this case, we want a `stat` struct. We can just stack allocate it:

```swift
var sb: stat = stat()

withUnsafePointer(to: &sb) { sbPtr in
    // Do whatever
}
```

Reply:
thanks, something learned!